portality/view/forms.py

```diff
@@ -95,9 +95,25 @@ class EditorGroupForm(Form):
     group_id = HiddenField("Group ID", [validators.Optional()])
     name = StringField("Group Name", [validators.DataRequired(), UniqueGroupName()])
     maned = StringField("Managing Editor", [validators.Optional(), MustHaveRole("admin")])
-    editor = StringField("Editor", [validators.DataRequired(), NotRole("publisher")])
+
+    editor = StringField("Editor", [validators.Optional(), NotRole("publisher")])
     associates = StringField("Associate Editors", [validators.Optional(), NotRole("publisher")])
 
+    def __init__(self, formdata=None, obj=None, prefix='', data=None, meta=None, **kwargs):
+        super().__init__(formdata=formdata, obj=obj, prefix=prefix, data=data, meta=meta, **kwargs)
+        self.editor_error = False;
+        self.maned_error = False;
+
+    def _validate_editors(self):
+        if not self.data["maned"]:
+            self.maned_error = True;
+        if not self.data["editor"]:
+            self.editor_error = True;
+        return not self.maned_error and not self.editor_error;
+
+    def validate(self, extra_validators = None):
+        return self._validate_editors() and super(EditorGroupForm, self).validate();
+
```

Comment:
There's a few oddities here.  

There are a bunch of semi-colons on the new code, which aren't necessary, looks like you were in javascript mode when you wrote this :)

Also, overriding the main validate function isn't something I've seen done before.  The validation effect we want to achieve is for both managing editor and editor to be required fields, so we just need to remove the Optional validator from both of those, surely?

Reply:
You're absolutely right—there's definitely something strange going on here! I must have had one foot in JavaScript and the other in... well, somewhere else entirely. 😅 Thanks for catching those rogue semi-colons.

As for the validation function override, I honestly have no idea what I was thinking there. I'll get that sorted out right away and remove the unnecessary override. Thanks for the sharp eye, and sorry for the code 💩 !